src/db/awsES/locations-dao.js

```diff
@@ -0,0 +1,191 @@
+    } else {
+      q.mustNot(esb.existsQuery('attributes.openHours'));
+    }
+  }
+
+  if (parsedParams.campus && parsedParams.campus.operator === 'oneOf') {
+    q.must(esb.termsQuery('attributes.campus', parsedParams.campus.value));
+  }
+
+  if (parsedParams.parkingZoneGroup && parsedParams.parkingZoneGroup.operator === 'oneOf') {
+    q.must(esb.termsQuery('attributes.parkingZoneGroup', parsedParams.parkingZoneGroup.value));
+  }
+
+  /*
+  geo distance query not working.
+  Maybe has to do with the naming of the lat and lon fields in index?
+
+  if (parsedParams.coordinates !== undefined) {
+    const latitude = parsedParams.coordinates[0];
+    const longitude = parsedParams.coordinates[1];
+    const distance = `${parsedParams.distance}${parsedParams.distanceUnit}`;
+    console.log(parsedParams.coordinates);
+    q.must(esb.geoDistanceQuery()
+      .field('attributes.geoLocation')
+      .distance(distance)
+      .geoPoint(esb.geoPoint().lat(latitude).lon(longitude)));
+  }
+  */
+
+  /* Will implement after GET /locations
+  if (queryParams['include'] !== undefined) {}
+  */
+  return esb.requestBodySearch().query(q).toJSON();
+};
+
+/**
+ * Return a list of Locations
+ * @param queryParams Object containing query parameters
+ * @returns {Promise<object>} Promise object represents a list of locations
+ */
+// GET /locations only returns 10 objects?
+const getLocations = async (queryParams) => {
+  const client = elasticsearch.Client({
+    host: domain,
+    log: 'error',
+    connectionClass,
+    awsConfig: new aws.Config({
+      accessKeyId,
+      secretAccessKey,
+      region: 'us-east-2',
+    }),
+  });
+
+  const res = await client.search({
+    index: 'locations',
+    body: buildQueryBody(queryParams),
+  });
+
+  const locations = [];
+  _.forEach(res.hits.hits, (location) => {
+    // eslint-disable-next-line dot-notation
+    const locationSource = location['_source'];
+    const locationAttributes = locationSource.attributes;
+    locationSource.attributes.abbreviations = {
+      arcGis: locationAttributes.arcgisAbbreviation,
+      banner: locationAttributes.bannerAbbreviation,
+    };
+    locationSource.attributes.giRestrooms = {
+      count: locationAttributes.girCount,
+      limit: locationAttributes.girLimit,
+      locations: (locationAttributes.girLocations)
+        ? locationAttributes.girLocations.split(', ')
+        : null,
+    };
+    locationSource.attributes.parkingSpaces = {
+      evSpaceCount: locationAttributes.evParkingSpaceCount,
+      adaSpaceCount: locationAttributes.adaParkingSpaceCount,
+      motorcyclesSpaceCount: locationAttributes.motorcycleParkingSpaceCount,
+    };
+    if (locationAttributes.geoLocation) {
+      locationSource.attributes.coordinates = {
+        lat: locationAttributes.geoLocation.latitude,
+        long: locationAttributes.geoLocation.longitude,
+      };
+    } else {
+      locationSource.attributes.coordinates = null;
```

Comment:
I think these pieces of logic should be moved to the serializer, what do you think?

Reply:
I agree